foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableDistOutputFolderForHabushuBuildCacheMigration.java

```diff
@@ -0,0 +1,57 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import com.boozallen.aissemble.upgrade.util.pom.PomModifications;
+import org.apache.maven.model.Build;
+import org.apache.maven.model.InputLocation;
+import org.apache.maven.model.Model;
+
+import java.io.File;
+
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.getLocationAnnotatedModel;
+import static com.boozallen.aissemble.upgrade.util.pom.PomHelper.writeModifications;
+
+
+/**
+ * This migration adds a new build directory tag to the build node of the project's pom.xml file, ensuring that Habushu
+ * modules leverage their dist folder for their output rather than target.  This approach aligns more closely to typical
+ * python conventions, and ensures that the build cache can find and replace those artifacts via the default config.
+ */
+public class EnableDistOutputFolderForHabushuBuildCacheMigration extends AbstractAissembleMigration {
+    private boolean isHabushuPackaged(Model model) {
+        return "habushu".equalsIgnoreCase(model.getPackaging());
+    }
+
+    private boolean doesBuildDirTagExist(Model model) {
+        return null != model.getBuild().getLocation("directory");
+    }
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        Model model = getLocationAnnotatedModel(file, getMavenProject());
+        return !doesBuildDirTagExist(model) && isHabushuPackaged(model);
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file, getMavenProject());
+        Build myMap = model.getBuild();
+        InputLocation buildLocation = myMap.getLocation("");
+        InputLocation inputLocation = new InputLocation(buildLocation.getLineNumber() + 1, buildLocation.getColumnNumber());
+        String line = "\t\t<directory>dist</directory>\n";
```

Comment:
A: Not a huge deal, but the `Insertion` modification will pass you the appropriate indent string based on the input file.  So if they use 4 spaces and you set indent to 2, it will pass 8 spaces to the content generator function.  If it's tabs then it passes two tabs. So your `Insertion` constructor would just change to `l -> l + line`.